packages/objection/src/utils.ts

```diff
@@ -0,0 +1,20 @@
+import { Model } from 'objection';
+
+function posixRegex(ignoreCase: boolean) {
+  const operator = ignoreCase ? '~*' : '~';
+  return `:field: ${operator} :regex`;
+}
+
+function regexp() {
+  return ':field: regexp :regex = 1';
+}
+const dialects: Record<string, (a: boolean) => string> = {
+  pg: posixRegex,
+  mysql: regexp,
+  oracledb: posixRegex,
+};
+
+export function generateRegexQuery(ignoreCase: boolean) {
+  const { client: { config } } = Model.knex();
+  return dialects[config.client](ignoreCase);
```

Comment:
Can you please throw an error if there is no function for knex client?

Something like “regent operator does not support ${client} database”

Reply:
gotcha